torch/distributed/rpc/constants.py

```diff
@@ -8,3 +8,5 @@
 
 # For ProcessGroupAgent.
 DEFAULT_NUM_SEND_RECV_THREADS = 4
+# Same default timeout as in c10d.
+DEFAULT_PROCESS_GROUP_TIMEOUT = _default_pg_timeout
```

Comment:
Does this mean `pg->recvAnySource` will throw an exception if no data is received on the wire for 30 minutes? It seems like we should default this to infinity to ensure `recvAnySource` never throws a timed out exception.

If what I mentioned above recvAnySource is true, then we shouldn't allow users to specify `process_group_timeout`, since they can just set it to a low value and this would cause listenLoop to terminate. For the unit test, we could have a private API `_set_process_group_timeout_for_testing` which sets an appropriate value for an internal flag that we check before calling `init_process_group` (process_group_timeout shouldn't be part of RpcBackendOptions).

Reply:
Yes, this is true. It can be seen by setting a short process_group_timeout such as 5s, allowing RPC init to complete, and then just waiting. Currently listenLoop will crash since there was no data sent over the wire. Should we bump this to be infinite?

Reply:
Filed https://github.com/pytorch/pytorch/issues/33107 to track this